code-generation/codegen-app.py

```diff
@@ -0,0 +1,40 @@
+from langchain_openai import ChatOpenAI
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_core.runnables import RunnablePassthrough
+
+import streamlit as st
+
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-m", "--model_url", default="http://0.0.0.0:8001/v1")
+args = parser.parse_args()
+
+st.title("Code Generation App")
+
+if "messages" not in st.session_state:
+    st.session_state["messages"] = [{"role": "assistant", 
+                                     "content": "How can I help you?"}]
+    
+for msg in st.session_state.messages:
+    st.chat_message(msg["role"]).write(msg["content"])
+
+llm = ChatOpenAI(base_url=args.model_url, 
+                 api_key="EMPTY",
+                 streaming=True)
+
+prompt = ChatPromptTemplate.from_template("""You are helpful AI assistant to write a code for a given {input}""")
```

Comment:
Do you think we can improve this prompt at all? Have you researched any prompt templates that might be a better fit for code gen?

Reply:
Thank you Michael for your feedbacks. I will address them accordingly.
Regarding prompt, I have included the basic prompt templates here. Will definitely experiment with couple of others and update the code.